mimic/model/flavor_objects.py

```diff
@@ -1,279 +1,107 @@
+                    self.flavors_store.append(flavor)
 
-class RackspaceOnMetalFlavor(Flavor):
-    """
-    A Rackspace onMetal flavor object representation
-    """
-    flavors = {"OnMetal Compute v1": {"id": "onmetal-compute1", "ram": 32768, "vcpus": 20,
-                                      "rxtx_factor": 10000, "disk": 32},
-               "OnMetal IO v1": {"id": "onmetal-io1", "ram": 131072, "vcpus": 40,
-                                 "rxtx_factor": 10000, "disk": 32},
-               "OnMetal Memory v1": {"id": "onmetal-memory1", "ram": 524288, "vcpus": 24,
-                                     "rxtx_factor": 10000, "disk": 32}}
-
-    def extra_specs_json(self):
+    def list_flavors(self, include_details, absolutize_url):
         """
-        Create a JSON-serializable data structure describing
-        ``OS-FLV-WITH-EXT-SPECS:extra_specs`` for an onMetal flavor.
+        Return a list of flavors with details.
         """
-        return {
-            "quota_resources": "instances=onmetal-compute-v1-instances,ram=onmetal-compute-v1-ram",
-            "class": "onmetal",
-            "policy_class": "onmetal_flavor"
-        }
+        if len(self.flavors_store) < 1:
+            self.create_flavors_list()
+        flavors = []
+        for flavor in self.flavors_store:
+            if self.region_name != "IAD" and isinstance(flavor, RackspaceOnMetalFlavor):
+                continue
+            if include_details:
+                flavors.append(flavor.detailed_json(absolutize_url))
+            else:
+                flavors.append(flavor.brief_json(absolutize_url))
+        result = {"flavors": flavors}
 
+        return dumps(result)
 
-class RackspacePerformance1Flavor(Flavor):
-    """
-    A Rackspace perfomance flavor object representation
-    """
-    flavors = {"1 GB Performance": {"id": "performance1-1", "ram": 1024, "vcpus": 1, "rxtx_factor": 200,
-                                    "disk": 20},
-               "2 GB Performance": {"id": "performance1-2", "ram": 2048, "vcpus": 2, "rxtx_factor": 400,
-                                    "disk": 40},
-               "4 GB Performance": {"id": "performance1-4", "ram": 4096, "vcpus": 4, "rxtx_factor": 800,
-                                    "disk": 40},
-               "8 GB Performance": {"id": "performance1-8", "ram": 8192, "vcpus": 8, "rxtx_factor": 1600,
-                                    "disk": 40}}
-
-    def extra_specs_json(self):
+    def get_flavor(self, http_get_request, flavor_id, absolutize_url):
         """
-        Create a JSON-serializable data structure describing
-        ``OS-FLV-WITH-EXT-SPECS:extra_specs`` for a performance1 flavor.
+        Return a flavor object if one exists from the list `/flavors` api,
+        else creates and adds the flavor to the :obj: `flavors_store`.
+        If the `flavor_id` is listed in `mimic.canned_responses.mimic_presets`,
+        then will return 404.
```

Comment:
Is this comment outdated and/or a typo? Looks like it returns 404 if the `flavor_id` is *not* listed in the `flavors_store`, where flavor types are defined in `mimic.model.flavors`.